crystal/kmeans.cr

```diff
@@ -40,46 +35,43 @@ def dist(x : Point, y : Point)
 end
 
 def average(xs : Array(Point))
-  xs.reduce { | x, y | x + y } / xs.size
+  xs.reduce { |x, y| x + y } / xs.size
 end
 
 def closest(x : Point, choices : Array(Point))
-  choices.min_by { | y | dist(x, y) }
+  choices.min_by { |y| dist(x, y) }
 end
 
 def clusters(xs : Array(Point), centroids : Array(Point))
-  xs.group_by { | x | closest(x, centroids) } .values
+  xs.group_by { |x| closest(x, centroids) }.values
 end
 
 def run(xs : Array(Point))
-  centroids = xs.first(n)
+  centroids = xs.first(N)
 
-  i = 0
-  while i < (iters - 1)
-    centroids = clusters(xs , centroids) .map { | l | average(l) }
-    i += 1
+  ITERS.times do
+    centroids = clusters(xs, centroids).map { |l| average(l) }
```

Comment:
Could be rewritten as `centroids = clusters(xs, centroids).map &->average(Array(Point))`.